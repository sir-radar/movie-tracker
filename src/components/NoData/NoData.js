import React from 'react';

const NoData = () => {
  return (
    <section className='no-data mt-5 mx-auto'>
      <h2 data-testid="nodata">No data to display</h2>
    </section>
  );
}

export default NoData;
