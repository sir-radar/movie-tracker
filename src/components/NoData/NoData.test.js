import React from 'react';
import { render, cleanup } from '@testing-library/react';
import NoData from './NoData';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer"

afterEach(cleanup)

const renderComponent = () => {
  const div = document.createElement("div");
  return render(<NoData/>, div)
}

it("renders without crashing", () => {
  renderComponent()
})

it("renders the router links", () => {
  const { getByTestId } = renderComponent()
  expect(getByTestId('nodata')).toHaveTextContent("No data to display")
})

it("matches snapshot", () => {
  const tree = renderer.create(<NoData/>).toJSON();
  expect(tree).toMatchSnapshot();
})