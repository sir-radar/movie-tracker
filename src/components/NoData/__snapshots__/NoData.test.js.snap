// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="no-data mt-5 mx-auto"
>
  <h2
    data-testid="nodata"
  >
    No data to display
  </h2>
</section>
`;
